### PR TITLE
oneDPL 2022.0.0 changes description

### PR DESCRIPTION
**Pull request by ValentinaKats**
Description of breaking changes in oneDPL 2022.0.0 release

### CODE REVIEW

File: documentation/library_guide/oneDPL_2022.0_changes.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
+oneDPL 2022.0 Changes
+#####################
+
+This page lists breaking changes in oneDPL 2022.0 from oneDPL 2021.
```
Comment (dcbenito) [inline documentation/library_guide/oneDPL_2022.0_changes.rst:4]: This page list the breaking changes that occurred between the 2021.0 and 2022.0 releases.

File: documentation/library_guide/oneDPL_2022.0_changes.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+Support of C++11 and C++14 has been discontinued.
```
Comment (dcbenito) [inline documentation/library_guide/oneDPL_2022.0_changes.rst:6]: Support for C++11 and C++14 has been discontinued.

File: documentation/library_guide/oneDPL_2022.0_changes.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+The internal parameter n of discard_block_engine was changed
+ from int to std::size_t type to align implementation with
+ С++ std proposal https://cplusplus.github.io/LWG/issue3561.
+ This change lets users to utilize the full range of values for P and R
+ template parameters of discard_block_engine.
+ 
+ .. Note::
+
+ Note: it may require rebuilding users’ code.
```
Comment (dcbenito) [inline documentation/library_guide/oneDPL_2022.0_changes.rst:16]: Note: You may need to rebuild your code.

File: documentation/library_guide/oneDPL_2022.0_changes.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+
+The following algorithms have been removed from dpl namespace
```
Comment (dcbenito) [inline documentation/library_guide/oneDPL_2022.0_changes.rst:19]: Can you add another layer of bullets and add colons? Change to: 

The following algorithms have been removed from the dpl namespace:
- include/oneapi/dpl/functional:
  - using ::std::binary_function;
  - using ::std::unary_function;

The following algorithms are deprecated in C++17 and removed for C++20:
- include/oneapi/dpl/functional:
  - using ::std::binary_negate;
  - using ::std::not1;
  - using ::std::not2;
  - using ::std::unary_negate;
- include/oneapi/dpl/type_traits:
  - using ::std::is_literal_type;
  - using ::std::is_literal_type_v;
  - using ::std::result_of;
  - using ::std::result_of_t;

File: documentation/library_guide/oneDPL_2022.0_changes.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
+oneDPL 2022.0 Changes
+#####################
+
+This page lists breaking changes in oneDPL 2022.0 from oneDPL 2021.
+
+Support of C++11 and C++14 has been discontinued. 
+
+The internal parameter n of discard_block_engine was changed
```
Comment (dcbenito) [inline documentation/library_guide/oneDPL_2022.0_changes.rst:8]: I made some minor edits. 

The internal parameter ``n`` of ``discard_block_engine`` was changed
 from a ``int`` to a ``std::size_t`` type to align its implementation with the
 С++ standard proposal found at https://cplusplus.github.io/LWG/issue3561.
 This change lets you utilize the full range of values for the ``P`` and ``R``
 template parameters of ``discard_block_engine``.

**Review by dcbenito — DISMISSED** [review]
I have updated with my changes. Please let me know if there are any questions or if I interpreted anything incorrectly.

File: documentation/library_guide/oneDPL_2022.0_changes.rst (hunk 1 above)
Comment (ValentinaKats) [reply documentation/library_guide/oneDPL_2022.0_changes.rst]: Applied the suggestion.

File: documentation/library_guide/oneDPL_2022.0_changes.rst (hunk 2 above)
Comment (ValentinaKats) [reply documentation/library_guide/oneDPL_2022.0_changes.rst]: Applied the suggestion.

File: documentation/library_guide/oneDPL_2022.0_changes.rst (hunk 3 above)
Comment (ValentinaKats) [reply documentation/library_guide/oneDPL_2022.0_changes.rst]: Applied the suggestion.

File: documentation/library_guide/oneDPL_2022.0_changes.rst (hunk 4 above)
Comment (ValentinaKats) [reply documentation/library_guide/oneDPL_2022.0_changes.rst]: Applied the suggestion.

File: documentation/library_guide/oneDPL_2022.0_changes.rst (hunk 5 above)
Comment (ValentinaKats) [reply documentation/library_guide/oneDPL_2022.0_changes.rst:7]: Applied the suggestion.

**Review by paveldyakov — DISMISSED** [review]

**Comment by dcbenito** [comment]
Can we merge this today? I am getting my docs ready for product validation and need to drop this doc by Monday of next week.

**Review by paveldyakov — DISMISSED** [review]

File: documentation/library_guide/oneDPL_2022.0_changes.rst [hunk 6]
```diff
@@ -0,0 +1,37 @@
+oneDPL 2022 Breaking Changes
+############################
+
+This page list the incompatible changes that occurred between the 2021.7.1 and 2022.0 releases.
```
Comment (dcbenito) [inline documentation/library_guide/oneDPL_2022.0_changes.rst:4]: Please change to "This page lists..."

File: documentation/library_guide/oneDPL_2022.0_changes.rst [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+.. note::
+   You may need to modify and/or rebuild your code when switching to oneDPL 2022.
+
+Support for C++11 and C++14 has been discontinued. To use any functionality of oneDPL 2022,
+a compiler that supports C++17 or newer version of the C++ language standard is required.
+
+The following APIs are not supported in C++17 and have been removed from ``namespace oneapi::dpl``:
+
+* In the ``<oneapi/dpl/functional>`` header
```
Comment (dcbenito) [inline documentation/library_guide/oneDPL_2022.0_changes.rst:14]: Please have a colon after "header".

File: documentation/library_guide/oneDPL_2022.0_changes.rst [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+  * ``binary_function``
+  * ``unary_function``
+
+The following APIs are deprecated in C++17 and not supported in C++20:
+
+* In the ``<oneapi/dpl/functional>`` header
```
Comment (dcbenito) [inline documentation/library_guide/oneDPL_2022.0_changes.rst:21]: Please have a colon after "header".

File: documentation/library_guide/oneDPL_2022.0_changes.rst [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  
+  * ``binary_negate``
+  * ``not1``
+  * ``not2``
+  * ``unary_negate``
+  
+* In the ``<oneapi/dpl/type_traits>`` header
```
Comment (dcbenito) [inline documentation/library_guide/oneDPL_2022.0_changes.rst:28]: Please have a colon after "header".

**Review by dcbenito — CHANGES_REQUESTED** [review]

File: documentation/library_guide/oneDPL_2022.0_changes.rst (hunk 7 above)
Comment (ValentinaKats) [reply documentation/library_guide/oneDPL_2022.0_changes.rst]: Done

File: documentation/library_guide/oneDPL_2022.0_changes.rst (hunk 9 above)
Comment (ValentinaKats) [reply documentation/library_guide/oneDPL_2022.0_changes.rst]: Done

File: documentation/library_guide/oneDPL_2022.0_changes.rst (hunk 6 above)
Comment (ValentinaKats) [reply documentation/library_guide/oneDPL_2022.0_changes.rst]: Done

File: documentation/library_guide/oneDPL_2022.0_changes.rst (hunk 8 above)
Comment (ValentinaKats) [reply documentation/library_guide/oneDPL_2022.0_changes.rst]: Done

**Review by dcbenito — APPROVED** [review]

**Comment by dcbenito** [comment]
LGTM. Thanks!

**Pull request merged**